flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js

```diff
@@ -1383,135 +1161,221 @@ class A320_Neo_CDU_MainDisplay extends FMCMainDisplay {
+        if (onlineFeaturesStatus === "UNKNOWN") {
+            CDU_OPTIONS_TELEX.ShowPage(this.fmc, this);
         }
+
+        this.initKeyboardScratchpad();
     }
 
-    deleteMessage(id) {
-        if (!this.messages[id]["opened"]) {
-            const cMsgCnt = SimVar.GetSimVarValue("L:A32NX_COMPANY_MSG_COUNT", "Number");
-            SimVar.SetSimVarValue("L:A32NX_COMPANY_MSG_COUNT", "Number", cMsgCnt <= 1 ? 0 : cMsgCnt - 1);
-        }
-        this.messages.splice(id, 1);
+    onUpdate(_deltaTime) {
+        super.onUpdate(_deltaTime);
+
+        this.mcdus.forEach((mcdu) => mcdu.onUpdate(_deltaTime));
+
+        this.lcdOverlay.style.opacity = SimVar.GetSimVarValue("L:A32NX_MFD_MASK_OPACITY", "number");
     }
 
-    // OUTGOING/SENT AOC MESSAGES
-    getSentMessages() {
-        return this.sentMessages;
+    initMcduVariables() {
+        this.mcdus.forEach((mcdu) => mcdu.clearMessageQueue());
+        this.aocAirportList.init();
     }
 
-    getSentMessage(id, type) {
-        const messages = this.sentMessages;
-        const currentMessageIndex = messages.findIndex(m => m["id"].toString() === id.toString());
-        if (type === 'previous') {
-            if (messages[currentMessageIndex - 1]) {
-                return messages[currentMessageIndex - 1];
+    /**
+     * Updates all MCDU displays, with optional exclude
+     */
+    requestUpdate(excludeMcdu) {
+        this.mcdus.forEach((mcdu) => {
+            if (mcdu !== excludeMcdu) {
+                mcdu.requestOnsideUpdate();
             }
-            return null;
-        } else if (type === 'next') {
-            if (messages[currentMessageIndex + 1]) {
-                return messages[currentMessageIndex + 1];
+        });
+    }
+
+    onEvent(_event) {
+        super.onEvent(_event);
+        const match = _event.match(/^([0-9])_BTN_(.+)$/);
+        if (match !== null) {
+            const mcduIndex = parseInt(match[1]) - 1;
+            const event = match[2];
+            if (mcduIndex >= 0 && mcduIndex < this.mcdus.length) {
+                this.mcdus[mcduIndex].onEvent(event);
             }
-            return null;
         }
-        return messages[currentMessageIndex];
     }
 
-    getSentMessageIndex(id) {
-        return this.sentMessages.findIndex(m => m["id"].toString() === id.toString());
+    onPowerOn() {
+        super.onPowerOn();
+    }
+
+    /**
+     * Add a new scratchpad message to all MCDUs (should be type II)
+     * @param message {{text, isAmber, isTypeTwo}} Message Object
+     * @param isResolved {function} Function that determines if the error is resolved at this moment (type II only).
+     * @param onClear {function} Function that executes when the error is actively cleared by the pilot (type II only).
+     */
+    addNewMessage(message, isResolved = () => {
+        return false;
+    }, onClear = () => {}) {
+        this.mcdus.forEach((mcdu) => mcdu.addNewMessage(message, isResolved, onClear, false));
     }
```

Comment:
It doesn't change anything, but since it returns false anyways, we can simplify the arrow function.

```suggestion
    addNewMessage(message, isResolved = () => false, onClear = () => {}) {
        this.mcdus.forEach((mcdu) => mcdu.addNewMessage(message, isResolved, onClear, false));
    }
```